phpseclib/Crypt/Common/SymmetricKey.php

```diff
@@ -2006,17 +2015,17 @@ private function setupMcrypt()
     protected function pad($text)
     {
         $length = strlen($text);
-
+        if ($length % $this->block_size == 0) {
+            return $text;
+        }
```

Comment:
phpseclib uses PKCS7 padding, when padding is enabled. What that means is that when you have a string whose length is a multiple of the block length you add chr(x) to the end of that string x times, where x is the block length. More info:

https://tools.ietf.org/html/rfc5652#section-6.3
https://en.wikipedia.org/wiki/Padding_(cryptography)#PKCS7

This commit breaks padding when the block size and the pad amount are the same.